File: gc-starter/crud-spring-boot-starter/src/main/java/com/gc/starter/crud/query/SortQueryParameter.java
package com.gc.starter.crud.query;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * 排序参数类
 * @author jackson
 * 2020/3/14 6:52 下午
 */
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class SortQueryParameter<K, V> extends QueryParameter<K, V> {

    private static final long serialVersionUID = -8278647627006548908L;

    private String sortName;

    private String sortOrder;
}

File: smart-starter/crud-spring-boot-starter/src/main/kotlin/com/smart/starter/crud/query/SortQueryParameter.kt
package com.smart.starter.crud.query

import com.smart.starter.crud.model.BaseModel


/**
 *
 * @author ming
 * 2019/9/13 下午6:06
 */
open class SortQueryParameter<T: BaseModel> : QueryParameter<T>() {
    // 排序字段
    var sortName: String? = null
    // 排序方向
    var sortOrder: String? = null

    override fun toString(): String {
        return "${super.toString()}, sortName=$sortName, sortOrder=$sortOrder"
    }


}